tests/unit/amo/components/TestHeader.js

```diff
@@ -100,6 +100,28 @@ describe(__filename, () => {
     expect(link.prop('children')).toEqual('View My Collections');
   });
 
+  it('displays a view profile link when user is signed in', () => {
```

Comment:
Should we include tests to show that it does not display the links when there is no signed in user?

Reply:
When user is not signed in, I think it should return a 404 and that's what I have done in the new iteration (not pushed yet).

Reply:
Maybe an [`AuthenticateButton`](https://github.com/mozilla/addons-frontend/blob/master/src/core/components/AuthenticateButton/index.js) would be even better than a 404.

Reply:
Also, this is about the menu, so the 404 or AuthenticateButton aren't really relevant to this - it's just about showing a link or not, right?

Reply:
The legacy frontend redirects the user to FxA so a login button might make sense, but the page will be very much empty.

Reply:
Well the page should not be displayed at all if there is no current logged in user IMO, because we do not really want to show a form in this case.